src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/AbstractTypeImportCompletionProvider.cs

```diff
@@ -78,79 +82,122 @@ private async Task AddCompletionItemsAsync(CompletionContext completionContext,
             var document = completionContext.Document;
             var project = document.Project;
             var workspace = project.Solution.Workspace;
-            var typeImportCompletionService = workspace.Services.GetService<ITypeImportCompletionService>();
+            var typeImportCompletionService = document.GetLanguageService<ITypeImportCompletionService>();
 
             // Find all namespaces in scope at current cursor location, 
             // which will be used to filter so the provider only returns out-of-scope types.
             var namespacesInScope = GetNamespacesInScope(document, syntaxContext, cancellationToken);
 
+            var tasksOfGettingTopLevelTypes = new ConcurrentSet<Task<Task>>();
```

Comment:
`Task<Task>`?

Reply:
I have broken the code that getting all the completion items from a refrence into two parts. First part populates the cache if needed, and the second part, which is returned as a not yet started Task, adds those items to CompletionContext. The items will be added only when the cache is ready within specified time out period.

This is because I don't want to show items from some references but not others, in case of timeout.

Reply:
> which is returned as a not yet started Task, adds those items to CompletionContext. The items will be added only when the cache is ready within specified timeout period.

Why is that a task...

WHy not just get the values and just add them.  I don't get how there's a need for asynchrony there...

Reply:
> WHy not just get the values and just add them. I don't get how there's a need for asynchrony there...

Yea, you are right. Will fix. Thanks! :)

Reply:
Fixed